llvm/test/CodeGen/AMDGPU/preload-implicit-kernargs-debug-info.ll

```diff
@@ -1,7 +1,10 @@
-; RUN: not --crash opt -mtriple=amdgcn-amd-amdhsa -mcpu=gfx940 -passes='amdgpu-attributor,function(amdgpu-lower-kernel-arguments)' -amdgpu-kernarg-preload-count=16 -S < %s 2>&1 | FileCheck %s
+; RUN: opt -mtriple=amdgcn-amd-amdhsa -mcpu=gfx940 -passes='amdgpu-attributor,function(amdgpu-lower-kernel-arguments)' -amdgpu-kernarg-preload-count=16 -S < %s 2>&1 \
+; RUN: | FileCheck -implicit-check-not='declare {{.*}} !dbg' %s
```

Comment:
this is still a big aggressive for check-not, and I'm not sure it supports regex. Can you simplify to just check-not=declare and explicitly check the few declares that are expected?

Reply:
I had already double-checked that --implicit-check-not supports regex (just `--implicit-check-not='{{.*}}' will fail on the first line`)

I went that route to avoid the awkward CHECKs for the irrelevant declares of `llvm.amdgcn.implicitarg.ptr` and `llvm.amdgcn.kernarg.segment.ptr`, but I posted a version with `--implicit-check-not=declare`  and the extra CHECKs; let me know what you think

Reply:
For some reason pushing didn't clear your approval, but I wanted to double check your thoughts on the test change so I re-requested